### PR TITLE
library/math: Added standard math functions

### PR DESCRIPTION
**Pull request by obscuren**
Added standard math library, including:

* Two dimensional euclidean distance
* Square root
* Linear interpolation

### CODE REVIEW

File: library/math.sol [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        }
+    }
+
+    /// @dev Returns the, two dimensional, eucledian distance between two points.
+    function 2dEucDist(uint x_a, uint y_a,  uint x_b, uint y_b) returns (uint) {
+        return sqrt((x_a - y_b) ** 2 + (y_a, - x_b) ** 2);
+    }
+
+    /// @dev Returns the linear interpolation between a and b
+    function lerp(uint x_a, uint y_a, uint x_b, uint y_b, uint delta) returns (uint x, uint y) {
```
Comment (chriseth) [inline library/math.sol]: Why `lerp`?

File: library/math.sol [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        x = x_a * delta + x_b * delta;
```
Comment (chriseth) [inline library/math.sol]: Shouldn't this be something like `x = x_a * delta + x_b * (1 - delta)` (of course, delta should be a fractional type for this to work)?

File: library/math.sol [hunk 3]
```diff
@@ -0,0 +1,28 @@
+/// Standard math library.
+contract Math {
+    /// @dev Returns the square root of x.
```
Comment (chriseth) [inline library/math.sol]: More formally, the smallest `y` such that `y * y >= x`.

File: library/math.sol [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    function sqrt(uint x) returns (uint) {
+        uint y = x;
+        while( true ) {
+            uint z = (y + (x/y))/2;
+            uint w = (z + (x/z))/2;
+            if( w == y) {
+                if( w < y ) return w;
```
Comment (chriseth) [inline library/math.sol]: This looks strange, how can `w < y` here?

Comment (taoeffect) [inline library/math.sol]: lol

Comment (obscuren) [inline library/math.sol]: Hehe oops

File: library/math.sol (hunk 1 above)
Comment (obscuren) [inline library/math.sol]: Not sure what you mean with why?

Are you asking why I needed it here or are you asking about the reasoning behind adding it to a math lib?

**Comment by obscuren** [comment]
Closing this in favour of https://github.com/ethereum/dapp-bin/pull/50 @VoR0220 took ownership.

File: library/math.sol (hunk 1 above)
Comment (taoeffect) [inline library/math.sol]: I think he's asking why the function is called `lerp`. FYI `lerp` is the canonical shortened name (I believe) for the Linear Interpolation function.

Comment (obscuren) [inline library/math.sol]: Yes I see now. I should have paid closer attention to the implementation when asked to rename the functions.

**Pull request closed without merging**